# Exercice 2
### Firas Kahlaoui 3 BD 1

In [33]:
library(readxl)
library(nnet)
library(tseries)
library(vars)

Warning message:
"package 'vars' was built under R version 4.4.2"
Loading required package: MASS

Loading required package: strucchange

Warning message:
"package 'strucchange' was built under R version 4.4.2"
Loading required package: zoo


Attaching package: 'zoo'


The following objects are masked from 'package:base':

    as.Date, as.Date.numeric


Loading required package: sandwich

Warning message:
"package 'sandwich' was built under R version 4.4.2"
Loading required package: urca

Warning message:
"package 'urca' was built under R version 4.4.2"
Loading required package: lmtest



In [20]:
# Read the Excel file
data <- read_excel("./data/exercice2.xlsx")
# Display the first few rows of the data
head(data)
str(data)

Continente Dis_Mag_Value nombre_total_de_deces Total_Affecte
1 2          8               12                      40       
2 3          7              476                  141277       
3 2          7                3                      15       
4 4          7             1250                  134250       
5 3          7               31                     200       
6 1          8              100                    2000       
  Total_des_dommages
1 500000            
2 100000            
3   7310            
4   6000            
5   3150            
6  25000

tibble [108 × 5] (S3: tbl_df/tbl/data.frame)
 $ Continente           : num [1:108] 2 3 2 4 3 1 2 4 3 2 ...
 $ Dis_Mag_Value        : num [1:108] 8 7 7 7 7 8 7 6 6 8 ...
 $ nombre_total_de_deces: num [1:108] 12 476 3 1250 31 100 400 2 38 4 ...
 $ Total_Affecte        : num [1:108] 40 141277 15 134250 200 ...
 $ Total_des_dommages   : num [1:108] 500000 100000 7310 6000 3150 25000 125000 2000 8000 400 ...


## Expliquer la variable continente

In [21]:
model <- multinom(Continente ~ Dis_Mag_Value + nombre_total_de_deces
                  + Total_Affecte + Total_des_dommages, data = data)

summary(model)

# weights:  30 (20 variable)
initial  value 173.819295 
iter  10 value 165.148541
iter  20 value 131.900368
iter  30 value 125.614527
iter  40 value 125.098755
iter  50 value 124.569999
iter  60 value 123.846741
iter  70 value 122.842655
iter  80 value 122.716010
final  value 122.714668 
converged


Call:
multinom(formula = Continente ~ Dis_Mag_Value + nombre_total_de_deces + 
    Total_Affecte + Total_des_dommages, data = data)

Coefficients:
  (Intercept) Dis_Mag_Value nombre_total_de_deces Total_Affecte
2   -4.807929     0.6177011         -3.017864e-05  9.305018e-07
3    5.326481    -0.9018543         -3.933559e-04  1.298160e-06
4    7.680986    -1.4427079         -1.203217e-05  1.060519e-06
5   -6.456925     0.8655634         -2.439808e-02 -1.408477e-04
  Total_des_dommages
2       4.043856e-07
3       2.230761e-07
4       2.370130e-07
5      -1.795839e-04

Std. Errors:
   (Intercept) Dis_Mag_Value nombre_total_de_deces Total_Affecte
2 3.457665e-08  2.398805e-07          3.356182e-05  7.155007e-07
3 9.207998e-08  6.071898e-07          3.224619e-04  1.013857e-06
4 1.374329e-08  8.091810e-08          4.799140e-05  1.007641e-06
5 3.854395e-08  2.675433e-07          4.320014e-07  2.195122e-04
  Total_des_dommages
2       3.148590e-07
3       3.271382e-07
4       3.320683e-07
5    

### Interprétation du modèle

- **Dis Mag Value** : Le coefficient pour "Dis Mag Value" est positif pour la classe 2, négatif pour les classes 3 et 4. Cela suggère que "Dis Mag Value" a une influence variable sur "Continente", affectant positivement certaines classes et négativement d'autres.
- **Nombre total de décès** : Le coefficient est très faible pour toutes les classes, indiquant une influence marginale de cette variable sur "Continente".
- **Total Affecté** : Le coefficient est très faible, mais légèrement positif pour la classe 2 et négatif pour la classe 5, ce qui suggère une influence faible mais présente sur "Continente".
- **Total des dommages** : Les coefficients sont proches de zéro, ce qui indique une faible influence sur "Continente" pour cette variable.

## Déterminer la qualité du modèle

In [22]:
# Check model performance with a confusion matrix
predictions <- predict(model, newdata = data)
confusion_matrix <- table(predictions, data$Continente)
print(confusion_matrix)

           
predictions  1  2  3  4  5
          1 29 18 13  6  2
          2  7 13  3  1  0
          3  5  1  6  2  0
          4  0  0  0  2  0
          5  0  0  0  0  0



### Matrice de confusion :

- **Classe 1 :**
    - 29 prédictions correctes pour la classe 1.
    - 18 erreurs où la classe 1 a été classée comme classe 2.
    - 13 erreurs où la classe 1 a été classée comme classe 3.
    - 6 erreurs où la classe 1 a été classée comme classe 4.
    - 2 erreurs où la classe 1 a été classée comme classe 5.

- **Classe 2 :**
    - 13 prédictions correctes pour la classe 2.
    - 7 erreurs où la classe 2 a été classée comme classe 1.
    - 3 erreurs où la classe 2 a été classée comme classe 3.
    - 1 erreur où la classe 2 a été classée comme classe 4.
    - Aucune erreur où la classe 2 a été classée comme classe 5.

- **Classe 3 :**
    - 6 prédictions correctes pour la classe 3.
    - 5 erreurs où la classe 3 a été classée comme classe 1.
    - 1 erreur où la classe 3 a été classée comme classe 2.
    - 2 erreurs où la classe 3 a été classée comme classe 4.
    - Aucune erreur où la classe 3 a été classée comme classe 5.

- **Classe 4 :**
    - 2 prédictions correctes pour la classe 4.
    - Aucune erreur où la classe 4 a été classée comme classe 1.
    - Aucune erreur où la classe 4 a été classée comme classe 2.
    - Aucune erreur où la classe 4 a été classée comme classe 3.
    - 2 erreurs où la classe 4 a été classée comme classe 5.

- **Classe 5 :**
    - Aucune prédiction correcte pour la classe 5.
    - Aucune erreur où la classe 5 a été classée comme une autre classe.

In [23]:
# Calculate accuracy
accuracy <- sum(diag(confusion_matrix)) / sum(confusion_matrix)
print(paste("Accuracy:", accuracy))

[1] "Accuracy: 0.462962962962963"


### Précision du modèle :

- **Précision totale** : 46.3 % (0.463), ce qui est relativement faible.
- **Le modèle a des difficultés à prédire correctement la classe 5**, ce qui suggère que cette classe est sous-représentée ou difficile à distinguer des autres classes.
- **Les erreurs de classification entre les classes 1, 2 et 3 sont fréquentes**, ce qui montre que le modèle a du mal à différencier ces classes.


##  Expliquer la variable Dis Mag value

In [24]:
model2 <- multinom(Dis_Mag_Value ~  nombre_total_de_deces
                   + Total_Affecte + Total_des_dommages, data = data)

summary(model2)

# weights:  30 (20 variable)
initial  value 193.510023 
iter  10 value 178.341875
iter  20 value 133.918263
iter  30 value 127.265709
iter  40 value 126.230292
iter  50 value 125.705007
iter  60 value 125.098559
iter  70 value 124.351402
iter  80 value 124.027365
iter  90 value 123.578972
iter 100 value 123.309541
final  value 123.309541 
stopped after 100 iterations


Call:
multinom(formula = Dis_Mag_Value ~ nombre_total_de_deces + Total_Affecte + 
    Total_des_dommages, data = data)

Coefficients:
  (Intercept) nombre_total_de_deces Total_Affecte Total_des_dommages
4   1.4144650         -0.5933362445  6.511416e-06       0.0003920540
5   0.2818687         -0.0007285482  1.874325e-05       0.0003997089
6   1.3400011          0.0001790053  4.001649e-05       0.0003993648
7   1.7559870          0.0001754416  4.003299e-05       0.0003994797
8   0.4755223          0.0002000724  4.225506e-05       0.0003992066

Std. Errors:
   (Intercept) nombre_total_de_deces Total_Affecte Total_des_dommages
4 6.931299e-09          9.498022e-09  1.351225e-04       1.860743e-05
5 2.678236e-09          3.222084e-07  9.039424e-05       7.912646e-06
6 4.078630e-09          1.940440e-05  8.802337e-05       7.915000e-06
7 6.059144e-09          1.818987e-05  8.802112e-05       7.913400e-06
8 1.872733e-09          1.364613e-05  8.802322e-05       7.917249e-06

Residual Deviance

### Interprétation des résultats

**nombre_total_de_deces :**

- **Catégorie 4 :** Coefficient négatif (-0.5933) suggère que lorsque le nombre total de décès augmente, la magnitude des catastrophes pour cette catégorie diminue.
- **Catégories 5 à 8 :** Coefficients proches de 0, indiquant une faible relation entre le nombre total de décès et la magnitude dans ces catégories.

**Total_Affecte :**

- Coefficients positifs pour toutes les catégories, montrant qu'un plus grand nombre de personnes affectées est associé à une augmentation de la magnitude des catastrophes.

**Total_des_dommages :**

- Coefficients positifs pour toutes les catégories, indiquant qu'une augmentation des dommages financiers est associée à une magnitude plus élevée.

## Déterminer la qualité du modèle

In [25]:
# Check model performance with a confusion matrix
predictions2 <- predict(model2, newdata = data)
confusion_matrix2 <- table(predictions2, data$Dis_Mag_Value)
print(confusion_matrix2)

            
predictions2  1  4  5  6  7  8
           1  0  0  0  0  0  0
           4  0  0  0  0  0  0
           5  0  0  1  0  0  0
           6  0  0  0  0  0  0
           7  1  1  7 29 48 14
           8  0  0  0  1  0  6


### Matrice de confusion :
**Classe 1 :**

- Aucune prédiction correcte ni incorrecte (aucune donnée associée à cette classe).

**Classe 4 :**

- Aucune prédiction correcte ni incorrecte (aucune donnée associée à cette classe).

**Classe 5 :**

- 1 prédiction correcte.
- Pas d'erreurs pour cette classe.

**Classe 6 :**

- Aucune prédiction correcte ni incorrecte.

**Classe 7 :**

- 29 prédictions correctes.
- 7 mal classées comme 5, 48 comme 8, 1 comme 1, et 1 comme 4.

**Classe 8 :**

- 6 prédictions correctes.
- 1 mal classée comme 6.

In [26]:
# Calculate accuracy
accuracy2 <- sum(diag(confusion_matrix2)) / sum(confusion_matrix2)
print(paste("Accuracy:", accuracy2))

[1] "Accuracy: 0.509259259259259"


### Précision globale

**Accuracy : 0.509 (environ 51 %) :**

- **Le modèle a correctement classé environ 51 % des observations.**
- **Les erreurs de classification sont principalement concentrées dans les classes 7 et 8.**

##  Etudier la stationarité des variables:  Dis Mag value,  nombre total de décès,  total affecté et total des dommages

### *Stationarité : Dis Mag Value*

In [29]:
stastionarity_Dis_Mag_Value <- adf.test(data$Dis_Mag_Value, alternative = "stationary")
print(stastionarity_Dis_Mag_Value)

Warning message in adf.test(data$Dis_Mag_Value, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$Dis_Mag_Value
Dickey-Fuller = -4.2359, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



**p-value = 0.01 < 0.05, on peut conclure que Dis Mag Value est stationnaire d'ordre 0.**

### *Stationarité : Nombre total de décès*

In [30]:

stastionarity_nombre_total_de_deces <- adf.test(data$nombre_total_de_deces, alternative = "stationary")
print(stastionarity_nombre_total_de_deces)

Warning message in adf.test(data$nombre_total_de_deces, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$nombre_total_de_deces
Dickey-Fuller = -4.8164, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



**p-value = 0.01 < 0.05, on peut conclure que nombre total de deces est stationnaire d'ordre 0.**

### *Stationarité : Total Affecté*

In [31]:

stastionarity_Total_Affecte <- adf.test(data$Total_Affecte, alternative = "stationary")
print(stastionarity_Total_Affecte)

Warning message in adf.test(data$Total_Affecte, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$Total_Affecte
Dickey-Fuller = -4.0487, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



**p-value = 0.01 < 0.05, on peut conclure que Total Affecte est stationnaire d'ordre 0.**

### *Stationarité : Total des dommages*

In [32]:
stastionarity_Total_des_dommages <- adf.test(data$Total_des_dommages, alternative = "stationary")
print(stastionarity_Total_des_dommages)

Warning message in adf.test(data$Total_des_dommages, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$Total_des_dommages
Dickey-Fuller = -5.2938, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



**p-value = 0.01 < 0.05, on peut conclure que Total des dommages est stationnaire d'ordre 0.**

### Expliquer la variable Dis Mag value en fonction des variables nombre total de décès, total affecté et total des dommages.

**Toutes les variables, à savoir Dis Mag Value, nombre total de décès, total affecté et total des dommages, sont stationnaires d'ordre 0 (p-value < 0.05). Alors, on va utiliser le modèle VAR.**

In [34]:
# Select optimal lag
lag_selection <- VARselect(data, lag.max = 10)
print(lag_selection)
optimal_lag <- lag_selection$selection["AIC(n)"]

$selection
AIC(n)  HQ(n)  SC(n) FPE(n) 
     1      1      1      1 

$criteria
                  1            2            3            4            5
AIC(n) 7.596636e+01 7.610007e+01 7.623456e+01 7.648174e+01 7.674301e+01
HQ(n)  7.628643e+01 7.668687e+01 7.708809e+01 7.760199e+01 7.812999e+01
SC(n)  7.675767e+01 7.755082e+01 7.834474e+01 7.925135e+01 8.017205e+01
FPE(n) 9.819799e+32 1.126946e+33 1.302127e+33 1.699217e+33 2.276689e+33
                  6            7            8            9           10
AIC(n) 7.699193e+01 7.723591e+01 7.737594e+01 7.747201e+01 7.756892e+01
HQ(n)  7.864563e+01 7.915634e+01 7.956309e+01 7.992589e+01 8.028953e+01
SC(n)  8.108040e+01 8.198381e+01 8.278327e+01 8.353877e+01 8.429511e+01
FPE(n) 3.061651e+33 4.181450e+33 5.280749e+33 6.590796e+33 8.572216e+33

